qiskit_machine_learning/neural_networks/circuit_qnn.py

```diff
@@ -176,10 +154,17 @@ def _sample(self, input_data: np.ndarray, weights: np.ndarray) -> np.ndarray:
         self.quantum_instance.backend_options['memory'] = orig_memory
 
         # return samples
-        return np.array([self._interpret_bitstring(b) for b in result.get_memory()])
+        memory = result.get_memory()
+        samples = np.zeros((1, *self.output_shape))
+        for i, b in enumerate(memory):
+            sample: Union[int, Tuple[int, ...]] = int(b, 2)
+            if self._interpret:
+                sample = self._interpret(cast(int, sample))
```

Comment:
I see this pattern in here a bit. It might have been easier to have set the method on the class called _interpret, when given an interpret of None (default), to some "dummy callback" then simply returned the same value as it was given that way the self._interpret could have always been called and any one doing any future changes could not forget to do the if test before calling it as that would not longer be needed,

Reply:
adjusted accordingly